src/common/dpycmn.cpp

```diff
@@ -125,52 +127,70 @@ wxDisplay::wxDisplay(const wxWindow* window)
 // ----------------------------------------------------------------------------
 // functions forwarded to wxDisplayImpl
 // ----------------------------------------------------------------------------
+bool wxDisplay::IsConnected() const
+{
+    wxCHECK_MSG( IsOk(), false, wxT("invalid wxDisplay object") );
+
+    return m_impl->IsConnected();
+}
 
 wxRect wxDisplay::GetGeometry() const
 {
     wxCHECK_MSG( IsOk(), wxRect(), wxT("invalid wxDisplay object") );
 
-    return m_impl->GetGeometry();
+    if ( m_impl->IsConnected() )
+        return m_impl->GetGeometry();
+    return wxRect();
```

Comment:
This is not critical, but I'd put the return in exceptional case first, for consistency with the precondition check above, i.e.

```suggestion
    if ( !m_impl->IsConnected() )
        return wxRect();
        
    return m_impl->GetGeometry();
```

(and the same for all the other functions, of course)

Reply:
Fixed in dd3e5e9.